packages/blue-sdk-ethers-liquidation/src/swap/curve.ts

```diff
@@ -0,0 +1,143 @@
+import { Address } from "@morpho-org/blue-sdk";
+import { Provider, ethers } from "ethers";
+import { LiquidationEncoder } from "../LiquidationEncoder";
+import CurveStableSwapNGABI from "../abi/CurveStableSwapNG.json";
+
+export async function getCurveWithdrawalAmount(
+  provider: Provider,
+  amount: bigint,
+  tokenIndex: number,
+  curvePool: Address,
+): Promise<bigint> {
+  const contract = new ethers.Contract(
+    curvePool,
+    CurveStableSwapNGABI,
+    provider,
+  );
+
+  /**
+   * @notice Calculate the amount received when withdrawing a single coin
+   * @param _burn_amount Amount of LP tokens to burn in the withdrawal
+   * @param i Index value of the coin to withdraw
+   * @return Amount of coin received
+   */
+  const result = await contract.calc_withdraw_one_coin!(amount, tokenIndex);
+  return result;
+}
+
+export async function getCurveSwapOutputAmountFromInput(
+  provider: Provider,
+  amount: bigint,
+  inputTokenIndex: number,
+  outputTokenIndex: number,
+  curvePool: Address,
+): Promise<bigint> {
+  const contract = new ethers.Contract(
+    curvePool,
+    CurveStableSwapNGABI,
+    provider,
+  );
+
+  /**
+   * @notice Calculate the current output dy given input dx
+   * @dev Index values can be found via the `coins` public getter method
+   * @param i Index value for the coin to send
+   * @param j Index value of the coin to receive
+   * @param dx Amount of `i` being exchanged
+   * @return Amount of `j` predicted
+   */
+  const result = await contract.get_dy!(
+    inputTokenIndex,
+    outputTokenIndex,
+    amount,
+  );
+  return result;
+}
+
+export async function getCurveSwapInputAmountFromOutput(
+  provider: Provider,
+  destAmount: bigint,
+  inputTokenIndex: number,
+  outputTokenIndex: number,
+  curvePool: Address,
+): Promise<bigint> {
+  const contract = new ethers.Contract(
+    curvePool,
+    CurveStableSwapNGABI,
+    provider,
+  );
```

Comment:
Typechain should have generated typed interface just for that, so you are type-safe

Check typechain config and run hardhat compile or just typechain